background.js

```diff
@@ -1,33 +1,75 @@
-const showMessage = async (message) => {
+const setLoadingState = async (active) => {
+  const activeCss = `.caal-loading {
+    position: absolute;
+    display: flex;
+    background: #85a94e;
+    padding: 1em;
+
+    transform: translateX(-50%);
+    left: 50%;
+    margin-top: 40vh;
+
+    z-index: 2000;
+
+    box-shadow: 0 var(--borderHeight) 0 0 var(--borderColor),0 .7rem .95rem .05rem var(--secondaryBorderColor);
+    border-radius: var(--borderRadius,.5rem);
+
+    color: #fff;
+    font-family: var(--globalSecondaryFont);
+    font-size: 3rem;
+    line-height: 1;
+    font-weight: 700;
+    text-shadow: 0 .1rem 0 rgba(0,0,0,.4);
+  }`;
+
+  const deactiveCss = `.caal-loading {
+    display: none;
+  }`;
+
+  browser.tabs.insertCSS({ code: active ? activeCss : deactiveCss });
+};
+
+const fetchField = async (url, field) => {
+  const response = await fetch(url);
+  const data = await response.json();
+  return data[field];
+};
+
+const getPGN = async (playerName, gameUrl) => {
+  const archives = await fetchField(
+    `https://api.chess.com/pub/player/${playerName}/games/archives`,
+    "archives"
+  );
+
+  for (let i = archives.length - 1; i >= 0; i--) {
+    games = await fetchField(archives[i], "games");
+    for (let j = games.length - 1; j >= 0; j--) {
+      if (games[j]["url"] === gameUrl) {
+        return games[j]["pgn"];
```

Comment:
Minor:

For `url` and `pgn` you can use regular property access:

```suggestion
      if (games[j].url === gameUrl) {
        return games[j].pgn;
```

Reply:
Habits from Python. I know this should be separated commit, but somehow this refactor was merged with network error handling commit https://github.com/zeraye/chess-analysis-firefox-extension/pull/2/commits/dc504f92764a4872d3370230eccd895f7315d168